src/timeseriesflattener/flattened_dataset.py

```diff
@@ -7,101 +7,157 @@ class FlattenedDataset:
     def __init__(
         self,
         prediction_times_df: DataFrame,
-        timestamp_colname: str = "timestamp",
-        id_colname: str = "dw_ek_borger",
+        id_col_name: str = "dw_ek_borger",
+        timestamp_col_name: str = "timestamp",
     ):
-        """Class containing a time-series flattened.
+        """Class containing a time-series, flattened. A 'flattened' version is a tabular representation for each prediction time.
+        A prediction time is every timestamp where you want your model to issue a prediction.
+
+        E.g if you have a prediction_times_df:
+
+        id_col_name | timestamp_col_name
+        1           | 2022-01-10
+        1           | 2022-01-12
+        1           | 2022-01-15
+
+        And a time-series of blood-pressure values as an outcome:
+        id_col_name | timestamp_col_name | blood_pressure_value
+        1           | 2022-01-09         | 120
+        1           | 2022-01-14         | 140
+
+        Then you can "flatten" the outcome into a new df, with a row for each of your prediction times:
+        id_col_name | timestamp_col_name | latest_blood_pressure_within_24h
+        1           | 2022-01-10         | 120
+        1           | 2022-01-12         | NA
+        1           | 2022-01-15         | 140
```

Comment:
So it is like a time-dependent merge.

Reply:
`lookahead_merge()` and `lookbehind_merge()` (lookbehind is better than lookback as  seems to better explain to be what the function intends to do.

And then the internal function can be a `_lookaround_merge`

Reply:
Yeah, that's probably better! Added an issue for it.

Reply:
I btw don't think the internal function `_lookaround_merge` what now called `_add_col` should not be internal. I could imagine that e.g. I would like to call that directly.

Reply:
That sentence is pretty hard to parse ;-)

I take it you think that `lookaround_merge` (which is currently called `add_col`) shouldn't be internal? Sure, I'll change that in main as well.

Reply:
Yep that is what I meant (y)